base-ledger-keyring.js

```diff
@@ -0,0 +1,623 @@
+const { EventEmitter } = require('events')
+const HDKey = require('hdkey')
+const ethUtil = require('ethereumjs-util')
+const sigUtil = require('eth-sig-util')
+const { TransactionFactory } = require('@ethereumjs/tx')
+
+const pathBase = 'm'
+const hdPathString = `${pathBase}/44'/60'/0'`
+const type = 'Ledger Hardware'
+
+const BRIDGE_URL = 'https://metamask.github.io/eth-ledger-bridge-keyring'
+
+const MAX_INDEX = 1000
+const NETWORK_API_URLS = {
+  ropsten: 'http://api-ropsten.etherscan.io',
+  kovan: 'http://api-kovan.etherscan.io',
+  rinkeby: 'https://api-rinkeby.etherscan.io',
+  mainnet: 'https://api.etherscan.io',
+}
+
+const CONNECTION_EVENT = 'ledger-connection-change'
+
+class BaseLedgerKeyring extends EventEmitter {
+  constructor (opts = {}) {
+    super()
+    this.accountDetails = {}
+    this.bridgeUrl = null
+    this.type = type
+    this.page = 0
+    this.perPage = 5
+    this.unlockedAccount = 0
+    this.hdk = new HDKey()
+    this.paths = {}
+    this.network = 'mainnet'
+    this.implementFullBIP44 = false
+    this.deserialize(opts)
+  }
+
+  /**
+   * Performs any asynchronous initialisation
+   *
+   * @returns {Promise<void>}
+   */
+  init () {
+    throw new Error('Method init not implemented')
+  }
+
+  destroy () {
```

Comment:
In the trezor keyring there's a similar method called `dispose`. It'd be a breaking change, but we could rename this one to `dispose` and only call it if it exists, instead of calling one or the other on the extension depending on the keyring type.